stackit/internal/services/iaas/networkinterfaceattach/resource.go

```diff
@@ -206,6 +252,10 @@ func (r *networkInterfaceAttachResource) Read(ctx context.Context, req resource.
 			if resp.Diagnostics.HasError() {
 				return
 			}
+
+			model.Id = utils.BuildInternalTerraformId(projectId, region, serverId, networkInterfaceId)
+			model.Region = types.StringValue(region)
+
```

Comment:
At the moment the values from model will not be written to the state, which will trigger a recreation when migrating to the new version

```suggestion
			// Set refreshed state
			resp.Diagnostics.Append(resp.State.Set(ctx, model)...)
			if resp.Diagnostics.HasError() {
				return
			}
```